### PR TITLE
Updating generate.ps1 files to consume powershell cmdlets

### PR DESCRIPTION
**Pull request by dsgouda**
Updating most of the generate.ps1 scripts except AzureStack and CognitiveServices

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — APPROVED** [review]
@dsgouda looks good. Let's get a review from @brjohnstmsft whose script have been changed significantly. Once we get a sign off, this can be merged.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@dsgouda can we delete the .cmd files at the same time while we are adding the .ps1 files for the one that you have changed. Rest of the repo will start adding .ps1 during PR review process.

**Review by bitsforthought — CHANGES_REQUESTED** [review]
Please do not delete the extra logic in `generate.ps1` for `Microsoft.Azure.Search.Data` and `Microsoft.Azure.Search.Service`. It is necessary to adjust the generated code such that we can build it the way we want.

We are considering ways to avoid this extra code-gen logic, for example by separating all our generated code from all our custom code, but that's a long-term effort for our team.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Fix-GeneratedCode.ps1
Comment (dsgouda) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Fix-GeneratedCode.ps1]: Will do.

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/generate.ps1
Comment (dsgouda) [reply src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/generate.ps1]: Absolutely, the name of the parameters remains the same too.

**Comment by dsgouda** [comment]
@brjohnstmsft addressed comments, PTAL

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/generate.ps1
Comment (brjohnstmsft) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Service/generate.ps1]: Please do not delete the extra logic in this file

**Review by bitsforthought — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@brjohnstmsft fixed now

**Review by bitsforthought — APPROVED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
